tests/rptest/tests/topic_recovery_test.py

```diff
@@ -345,7 +345,8 @@ class MissingPartition(BaseCase):
     only one key).
     """
 
-    topics = (TopicSpec(name='panda-topic',
+    topic_name = 'panda-topic'
+    topics = (TopicSpec(name=topic_name,
```

Comment:
rather than doing this, access the name from the topicspec: `self.topic` is a shortcut if there is exactly one topic, or `self.topics[n].name` if you have more than one.

Reply:
Thanks for the suggestion, I tried this on another PR to sneak in this change there, but it seems this property is only defined for `RedpandaTest` in `rptest.tests.redpanda_test.RedpandaTest.topic` whereas this code uses `rptest.tests.topic_recovery_test.MissingPartition` inherited from `rptest.tests.topic_recovery_test.BaseCase` neither of which have this property.

It might be good to define a similar property on this test case.